src/CMakeLists.txt

```diff
@@ -1064,6 +1064,11 @@ list( APPEND LIBRARIES
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:pthread>
 )
 
+# mingw needs these too
```

Comment:
This is better placed on the if condition above, but I am not familiar with cmake's `$<>` syntax, so went this way which also works.

Reply:
Do you know that the $<> things are called "generator expressions?"  Complete documentation here:  https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html?highlight=generator%20expressions#manual:cmake-generator-expressions(7)

In effect they make a small programming language of their own.  A first pass of configuration interprets CMakeLists.txt files, associating property values with targets.  A second pass over the targets interprets the generator expressions in the properties.  In this second pass only certain special variables are defined, not all of the variables that CMakeLists.txt uses in the first pass.